dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/ClassDumper.java

```diff
@@ -146,23 +146,22 @@ boolean isSystemClass(String className) {
   }
 
   /**
-   * Scans class files in the jar file and returns a {@link SymbolReferenceSet} populated with
-   * symbol references.
-   *
-   * @param jar absolute path to a jar file
+   * Returns a set of symbol references (maps from classes to class symbols, method symbols and
+   * field symbols) found in the class path.
    */
-  SymbolReferenceSet scanSymbolReferencesInJar(Path jar) throws IOException {
-    checkArgument(jar.isAbsolute(), "The input jar file path is not an absolute path");
-    checkArgument(Files.isReadable(jar), "The input jar file path is not readable");
+  SymbolReferenceMaps scanSymbolReferencesInClassPath() throws IOException {
+    SymbolReferenceMaps.Builder builder = new SymbolReferenceMaps.Builder();
 
-    SymbolReferenceSet.Builder symbolTableBuilder = SymbolReferenceSet.builder();
-    for (JavaClass javaClass : listClassesInJar(jar)) {
-      if (!isCompatibleClassFileVersion(javaClass)) {
-        continue;
+    for (Path jar : inputClassPath) {
+      for (JavaClass javaClass : listClassesInJar(jar)) {
+        if (!isCompatibleClassFileVersion(javaClass)) {
+          continue;
+        }
+        builder.addAll(scanSymbolReferences(jar, javaClass));
```

Comment:
Maybe we could pass in the ClassFile here, rather than the jar and javaClass? It seems weird to be passing a jar here. The ClassFile is really what w're scanning.

Reply:
That's a good idea. Updated.